controllers/members.js

```diff
@@ -135,7 +141,48 @@ const archiveMembers = async (req, res) => {
     return res.boom.notFound("User doesn't exist");
   } catch (err) {
     logger.error(`Error while retriving contributions ${err}`);
-    return res.boom.badImplementation(ERROR_MESSAGE);
+    return res.boom.badImplementation(SOMETHING_WENT_WRONG);
+  }
+};
+
+/**
+ * Purges the Cache of Members Profile Page
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const purgeMembersCache = async (req, res) => {
+  try {
+    const { id, username } = req.userData;
+
+    const response = await fetch(
+      CLOUDFLARE_PURGE_CACHE_API,
+      POST,
+      null,
+      { files: [`https://members.realdevsquad.com/${username}`] },
+      {
+        "X-Auth-Key": config.get("cloudflare.CLOUDFLARE_X_AUTH_KEY"),
+        "X-Auth-Email": config.get("cloudflare.CLOUDFLARE_X_AUTH_EMAIL"),
+      }
+    );
+
+    // eslint-disable-next-line no-console
+    console.log(response);
+
+    if (response.status === 200) {
+      await logsQuery.addLog(logType.CLOUDFLARE_CACHE_PURGED, { userId: id }, { message: "Cache Purged" });
+    } else {
+      await logsQuery.addLog(
+        logType.CLOUDFLARE_CACHE_PURGED,
+        { userId: id },
+        { message: `Error in Purging Cache - ${response}` }
+      );
+    }
+
+    return res.json({ message: "Cache purged successfully", ...response.data });
+  } catch (error) {
+    logger.error(`Error while clearing members cache: ${error}`);
+    return res.boom.badImplementation(SOMETHING_WENT_WRONG);
```

Comment:
For Discord Discussion - https://discord.com/channels/673083527624916993/1000471526174359663